raster/r.mapcalc/Makefile

```
@@ -20,7 +20,9 @@ default: multi
 $(BIN)/$(PGM2)$(EXE): LIBES = $(LIBES2)
 $(BIN)/$(PGM3)$(EXE): LIBES = $(LIBES3)
 
-$(OBJDIR)/mapcalc.yy.o: mapcalc.tab.h
+$(OBJDIR)/*.o: mapcalc.tab.h mapcalc.tab.c
+$(OBJDIR)/mapcalc.yy.o: mapcalc.tab.h mapcalc.tab.c
+
```

```suggestion
$(OBJDIR)/*.o $(OBJDIR)/mapcalc.yy.o: mapcalc.tab.h mapcalc.tab.c
```

While it bothers me not to be able to reproduce this issue (and to full depth understand what's going on), I think this solution is good. If my suggested simplification works I'd personally prefer to have this in one line (as they are logically interconnected). Or even better, would the following solve the issue

```
$(OBJDIR)/*.o: mapcalc.tab.h mapcalc.tab.c

```

---

Too late :)

---

@nilason I'm sorry, I just merged it (didn't understand that you wanted to keep it open and modify further). So, a new PR will be needed...

---

> @nilason I'm sorry, I just merged it (didn't understand that you wanted to keep it open and modify further). So, a new PR will be needed...

No worries! The one line solution I suggested should work (as it is basically the same). I can't test  the latter suggestion (just adding `mapcalc.tab.c ` as a pre-requisite to `$(OBJDIR)/*.o` targets) for solving the issue at hand, perhaps @bmwiedemann can chip in. The end-effect should be the same, but the code more concise.

---

`$(OBJDIR)/*.o: mapcalc.tab.h mapcalc.tab.c` alone did not work - my guess is that the wildcard only matches existing files. Combining both in one line should work - I'm giving it a test-run now and should have results in an hour.

---

> `$(OBJDIR)/*.o: mapcalc.tab.h mapcalc.tab.c` alone did not work - my guess is that the wildcard only matches existing files. Combining both in one line should work - I'm giving it a test-run now and should have results in an hour.

Thanks for testing and for your contribution!

---

`$(OBJDIR)/*.o $(OBJDIR)/mapcalc.yy.o: mapcalc.tab.h mapcalc.tab.c` worked fine.

---

> `$(OBJDIR)/*.o $(OBJDIR)/mapcalc.yy.o: mapcalc.tab.h mapcalc.tab.c` worked fine.

Thought so, thanks! I'll put up a PR, simplifying this.

Would you have been able to reproduce this issue you reported by only (re-)building in this module (directory)? E.g.: 
```
cd raster/r.mapcalc
make clean
make -j1
```
I'm asking because I try figure out my inability to reproduce.

---

[My tooling](https://github.com/bmwiedemann/reproducibleopensuse/blob/devel/rbk) does 2 whole package builds from scratch via [our spec file](https://code.opensuse.org/package/grass/blob/8a65e40d67fbd21ad4bcead967468190ee7a9f2c/f/grass.spec#_176). Not that easy to adapt to test a partial build. My guess is that there are some cross-dependencies from outside that dir that caused the issue.

---

> [My tooling](https://github.com/bmwiedemann/reproducibleopensuse/blob/devel/rbk) does 2 whole package builds from scratch via [our spec file](https://code.opensuse.org/package/grass/blob/8a65e40d67fbd21ad4bcead967468190ee7a9f2c/f/grass.spec#_176). Not that easy to adapt to test a partial build. My guess is that there are some cross-dependencies from outside that dir that caused the issue.

I did take a look at your tools, I'll see if I can make use of it on the Mac (definitely won't work as is).

Studying your log file, I realised the default GNU Make is quite old (3.81), but trying with 4.4.1 didn't make any difference.

Also I happened to notice:

```
 [   31s] configure: WARNING: unrecognized options: --enable-socket, --with-curses, --with-motif, --with-python, --with-wxwidgets
```

the listed options may all be removed from configure.

Thanks for your feedback!